v03_pipeline/lib/tasks/reference_data/updated_reference_dataset_collection_test.py

```diff
@@ -0,0 +1,328 @@
+                    enums=hl.Struct(primate_ai=hl.Struct(), cadd=hl.Struct()),
+                    date=ANY,
+                ),
+            ],
+        )
+
+    @mock.patch(
+        'v03_pipeline.lib.reference_data.dataset_table_operations.get_dataset_ht',
+    )
+    @mock.patch.object(ReferenceDatasetCollection, 'datasets')
+    def test_update_task_with_dataset_and_empty_reference_data_table(
+        self,
+        mock_reference_dataset_collection_datasets,
+        mock_get_dataset_ht,
+    ) -> None:
+        delete_reference_data_ht()
+        mock_reference_dataset_collection_datasets.return_value = ['primate_ai']
+        mock_get_dataset_ht.return_value = MOCK_PRIMATE_AI_DATASET_HT
+
+        worker = luigi.worker.Worker()
+        # create task with dataset parameter
+        task = UpdatedReferenceDatasetCollectionTask(
+            reference_genome=ReferenceGenome.GRCh38,
+            dataset_type=DatasetType.SNV_INDEL,
+            sample_type=SampleType.WGS,
+            reference_dataset_collection=ReferenceDatasetCollection.COMBINED,
+            dataset='primate_ai',
+        )
+        worker.add(task)
+        worker.run()
+        self.assertTrue(task.complete())
+
+        ht = hl.read_table(task.output().path)
+        self.assertCountEqual(
+            ht.collect(),
+            [
+                hl.Struct(
+                    locus=hl.Locus(
+                        contig='chr1',
+                        position=871269,
+                        reference_genome='GRCh38',
+                    ),
+                    alleles=['A', 'C'],
+                    primate_ai=hl.Struct(score=0.5),
+                ),
+            ],
+        )
+        self.assertEqual(
+            ht.globals.collect(),
+            [
+                hl.Struct(
+                    paths=hl.Struct(
+                        primate_ai='gs://seqr-reference-data/GRCh38/primate_ai/PrimateAI_scores_v0.2.liftover_grch38.ht',
+                    ),
+                    versions=hl.Struct(primate_ai='v0.2'),
+                    enums=hl.Struct(primate_ai=hl.Struct()),
+                    date=ANY,
+                ),
+            ],
+        )
+
+    @mock.patch(
+        'v03_pipeline.lib.tasks.reference_data.updated_reference_dataset_collection.UpdatedReferenceDatasetCollectionTask.initialize_table',
+    )
+    @mock.patch(
+        'v03_pipeline.lib.reference_data.dataset_table_operations.get_dataset_ht',
+    )
+    @mock.patch.object(ReferenceDatasetCollection, 'datasets')
+    def test_update_task_with_dataset_and_existing_reference_dataset_collection_table(
+        self,
+        mock_reference_dataset_collection_datasets,
+        mock_get_dataset_ht,
+        mock_initialize_table,
+    ) -> None:
+        delete_reference_data_ht()
+        # make initialize_table() return existing reference dataset collection table
+        mock_initialize_table.return_value = hl.Table.parallelize(
```

Comment:
We can just... not call `delete_reference_data_ht` and not mock here right?

Reply:
Ah.  I see.. we're ending up with weird stuff in here because our `def complete()` isn't implemented yet.  In that case, maybe not having a `setUp` and having `mock_initialize_table.return_value = hl.read_table(OMBINED_2_PATH)` here works?

Reply:
even so, since `complete()` is still the base class logic, if the file exists at `COMBINED_2_PATH`, initialize_table() won't run right?

Reply:
ah, I don't think so (at least, unless we're mocking the `valid_reference_dataset_collection` path).  `complete()` won't run if whatever `valid_reference_dataset_collection` points at exists... and that only exists if we `shutil.copy` it from the repo into the tmp dir.

Reply:
oh you're right!

Reply:
yeah, for some additional context on why it is this way...

we want to be able to test the pipeline under different reference dataset collection cases, but within the pipeline, there is only a single reference dataset collection (with a static path) per type.  That allows us to now have to pass a file path through the whole pipeline, like [v02 was doing](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/luigi_pipeline/seqr_loading.py#L46-L49)